### DISCUSSION TITLE
Consider migrating samples to CDK with Java

### DISCUSSION THREAD

**Discussion by maciejwalkowiak** [awspring/spring-cloud-aws#47] [General]
While writing CDK code (infrastructure for samples) in TypeScript is very convenient, it raises the bar for contributors to contribute samples. 

We should consider and give another try to CDK for Java. Ideally we would completely drop a need to install CDK and invoke CDK from the Maven plugin: https://github.com/LinguaRobot/aws-cdk-maven-plugin

I am not sure if this plugin is maintained - it deserves at least a CDK version bump. If not, we can fork it and maintain ourselves.

**Comment by eddumelendez** [comment]
CDK in Typescript is really good to start thinking about moving it to Java. As far as the project provide good examples others can benefit from that.

I agree CDK sometimes is not very clear (from my side) to know what I should sometimes use like `Cnf<Resource>` or `Resource` so saying that I can say that CLI is good.

**Reply by maciejwalkowiak** [reply]
Do you mean to move to Java or stay with TypeScript? Another issue that I see at the moment is updating dependencies. With Maven we could move all the versions to the parent pom and easily enforce all samples to use the same versions.

**Reply by eddumelendez** [reply]
I vote to keep using CDK with Typescript.
